### PR TITLE
[REVIEW] Move unary files to legacy

### PR DESCRIPTION
**Pull request by codereport**
[ISSUE] #2950

[Problem]
Need to refactor unary.hpp to use cudf::column instead of gdf

[Solution]
This is Phase I of of the unary.hpp refactor.
Moving the header and implementation files to the legacy
directories and updating corresponding code.

[Test]
Ran make test - 100% pass.

### CODE REVIEW

**Comment by codereport** [comment]
@harrism Looks like tests are running fine how after I made a fix to the #2971 merge issue. However, I also found `cudf/java/src/main/native/src/CudfJni.cpp:24:#include "cudf/unary.hpp"`

Does this need to be updated as well?

**Comment by rgsl888prabhu** [comment]
@codereport please resolve the conflicts

**Comment by codereport** [comment]
@rgsl888prabhu conflicts resolved, CI is rerunning

**Comment by codereport** [comment]
Thanks for the reviews @shwina and @dillon-cullinan. Is there anything left for me to do for this to be merged?

Also, #harrism (or anyone else), we should probably get a response on the post below before merging.

> @harrism Looks like tests are running fine now after I made a fix to the #2971 merge issue. However, I also found `cudf/java/src/main/native/src/CudfJni.cpp:24:#include "cudf/unary.hpp"`
> 
> Does this need to be updated as well?

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@
 - PR #3046 Define and implement new copying APIs `empty_like` and `allocate_like`
 - PR #3128 Support MultiIndex in DataFrame.join
 - PR #3021 Java host side concat of serialized buffers
+- PR #3138 Movey unary files to legacy
```
Comment (kkraus14) [inline CHANGELOG.md]:
```suggestion
- PR #3138 Move unary files to legacy
```

**Pull request merged**